rust/operator-binary/src/crd/mod.rs

```diff
@@ -246,6 +246,43 @@ impl v1alpha1::HiveCluster {
         format!("{name}-{role}", name = self.name_any(), role = hive_role)
     }
 
+    /// Set of functions to define service names on rolegroup level.
+    /// Headless service for cluster internal purposes only.
+    // TODO: Move to operator-rs
+    pub fn rolegroup_headless_service_name(
+        &self,
+        rolegroup: &RoleGroupRef<v1alpha1::HiveCluster>,
+    ) -> String {
+        format!("{name}-headless", name = rolegroup.object_name())
+    }
+
+    /// Headless metrics service exposes Prometheus endpoint only
+    // TODO: Move to operator-rs
+    pub fn rolegroup_headless_metrics_service_name(
+        &self,
+        rolegroup: &RoleGroupRef<v1alpha1::HiveCluster>,
+    ) -> String {
+        format!("{name}-metrics", name = rolegroup.object_name())
+    }
+
+    pub fn metrics_ports(&self) -> Vec<ServicePort> {
+        vec![ServicePort {
+            name: Some(METRICS_PORT_NAME.to_string()),
+            port: METRICS_PORT.into(),
+            protocol: Some("TCP".to_string()),
+            ..ServicePort::default()
+        }]
+    }
+
+    pub fn service_ports(&self) -> Vec<ServicePort> {
+        vec![ServicePort {
+            name: Some(HIVE_PORT_NAME.to_string()),
+            port: HIVE_PORT.into(),
+            protocol: Some("TCP".to_string()),
+            ..ServicePort::default()
+        }]
+    }
```

Comment:
This should not be implemented on the cluster (does not need self) and could go into a service.rs module.